### PR TITLE
Support for `"@import"`.

### PR DESCRIPTION
**Pull request by gkellogg**
It might not be playing nicely with caching, please consider.

### CODE REVIEW

File: lib/context.js [hunk 1]
```diff
@@ -266,6 +266,53 @@ api.process = async ({
...
+      if('@import' in importCtx) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; imported context must not include @import.',
+          'jsonld.SyntaxError',
+          {code: 'invalid context member', context: localCtx});
+      }
+
+      // merge ctx into importCtx and replace rval with the result
+      for(const key in importCtx) {
+        if(!(key in ctx)) {
```
Comment (davidlehn) [inline lib/context.js]: Avoid problem if importCtx has default Object property term names like 'toString':
```suggestion
        if(!ctx.hasOwnProperty(key)) {
```

File: lib/context.js [hunk 2]
```diff
@@ -266,6 +266,53 @@ api.process = async ({
...
+        base: options.base
+      });
+      if(resolvedImport.length !== 1) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import must reference a single context.',
+          'jsonld.SyntaxError',
+          {code: 'invalid remote context', context: localCtx});
+      }
+      const importCtx = resolvedImport[0].document;
+      if('@import' in importCtx) {
```
Comment (davidlehn) [inline lib/context.js]: Seems like this would need to be more involved to catch `@import` in scoped contexts too. There are not test for this.

Comment (gkellogg) [reply lib/context.js]: Catching it was going to require adding more state to be passed down, and even though it isn't the use case, it doesn't really cause any harm if its (mis)used this way. But, that would be an issue on the spec.

**Comment by davidlehn** [comment]
Does look like maybe a caching issue could happen.  I'm not sure what the situation is or if it's worth dealing with.  Maybe if you use the same resources as a context and also as an import?  Let's leave that alone until someone can figure out a test case.

**Pull request merged**